src/development/accessibility-and-localization/internationalization.md

```diff
@@ -264,9 +326,205 @@ return MaterialApp(
+    // Returns 'they'
+    Text(AppLocalizations.of(context)!.pronoun('other')),
+  ],
+);
+```
+
+### Messages with numbers and currencies
+Numbers and numbers that represent currency values are
+displayed very differently in different locales. 
+The localizations generation tool in `flutter_localizations` makes use of the
+[`intl` package's `NumberFormat` class](https://api.flutter.dev/flutter/intl/NumberFormat-class.html)
+to properly format numbers based on the locale and the
+desired format.
+
+The “format” for placeholders whose type is `int`, `double`, or `number`
+can be any one of the following `NumberFormat` constructors.
+
+<div class="table-wrapper" markdown="1">
+| Message "format" value      | Output for 1200000 |
+| --------------------------- | ------------------ |
+| `compact`                   | "1.2M"             |
+| `compactCurrency`*          | "$1.2M"            |
+| `compactSimpleCurrency`*    | "$1.2M"            |
+| `compactLong`               | "1.2 million"      |
+| `currency`*                 | "USD1,200,000.00"  |
+| `decimalPattern`            | "1,200,000"        |
+| `decimalPercentPattern`*    | "120,000,000%"     |
+| `percentPattern`            | "120,000,000%"     |
+| `scientificPattern`         | "1E6"              |
+| `simpleCurrency`*           | "$1,200,000"       |
+{:.table.table-striped}
+</div>
+
+The five starred `NumberFormat` constructors have optional, named parameters. Those
+parameters can be specified as the value of the placeholder’s “optionalParameters” object.
+For example, to specify the optional decimalDigits parameter for "compactCurrency":
+
+<?code-excerpt "gen_l10n_example/lib/l10n/app_en.arb" skip="34" take="13" replace="/{{/{{ '{{' }}/g;/},$/}/g"?>
+```json
+"numberOfDataPoints": "Number of data points: {value}",
+"@numberOfDataPoints": {
+  "description": "A message with a formatted int parameter",
+  "placeholders": {
+    "value": {
+      "type": "int",
+      "format": "compactCurrency",
+      "optionalParameters": {
+        "decimalDigits": 2
+      }
+    }
+  }
+}
+```
+
+### Messages with dates
+Dates strings are formatted in many different ways depending both the locale and the app’s needs.  
+
+Placeholder values with type `DateTime` are formatted with
+[`intl`'s `DateFormat` class](https://api.flutter.dev/flutter/intl/DateFormat-class.html).
+There are 41 format variations, identified by the names of their `DateFormat` factory constructors.
+In the following example, the `DateTime` value that appears in the `helloWorldOn` message is formatted with `DateFormat.yMd`:
+
+```json
+"helloWorldOn": "Hello World on {date}",
+"@helloWorldOn": {
+  "description": "A message with a date parameter",
+  "placeholders": {
+    "date": {
+      "type": "DateTime",
+      "format": "yMd"
+    }
+  }
+}
+```
+
+In an app, when the locale is US English, the following expression would produce  “7/10/1996”. If the locale was Russian, then it would produce “10.07.1996”.
```

Comment:
```suggestion
In an app where the locale is US English, the following expression would produce  “7/10/1996”. In a Russian locale, it would produce “10.07.1996”.
```